Tests/MessagingPush/PushHandling/AutomaticPushClickHandlingIntegrationTest.swift

```diff
@@ -0,0 +1,212 @@
+@testable import CioInternalCommon
+@testable import CioMessagingPush
+import Foundation
+import SharedTests
+import XCTest
+
+class AutomaticPushClickHandlingIntegrationTest: IntegrationTest {
+    private var pushEventHandler: PushEventHandler!
+
+    private let pushClickHandler = PushClickHandlerMock()
+    private let pushEventHandlerProxy = PushEventHandlerProxyImpl()
+
+    override func setUp() {
+        super.setUp()
+
+        pushEventHandler = iOSPushEventListener(
+            jsonAdapter: diGraph.jsonAdapter,
+            pushEventHandlerProxy: pushEventHandlerProxy,
+            moduleConfig: MessagingPushConfigOptions(),
+            pushClickHandler: pushClickHandler,
+            pushHistory: diGraph.pushHistory,
+            logger: diGraph.logger
+        )
+    }
+
+    // MARK: When the CIO SDK is the only click handler in the app.
+
+    func test_givenCioSdkOnlyPushHandler_givenClickedOnCioPush_expectEventHandled() {
+        let givenPush = PushNotificationStub.getPushSentFromCIO()
+
+        performPushAction(PushNotificationActionStub(push: givenPush, didClickOnPush: true))
+
+        assertHandledClick(for: givenPush)
+    }
+
+    // TODO: Handle handle scenario where a push gets clicked that is not a CIO push, and there are no push click handlers installed in an app to handle that push.
+    func test_givenCioSdkOnlyPushHandler_givenClickedOnPushNotSentFromCio_expectEventHandled() {
```

Comment:
are we expecting event to be handled? if given `givenClickedOnPushNotSentFromCio` ?

Reply:
`assertNoPushHanding()` says differently

Reply:
I think that your question is coming from the function name, `..._expectEventHandled`. 

if that's correct, I am OK with changing the name. 

Does the function body describe the behavior that we expect? 
* Click a push not sent from CIO
* Expect that the CIO SDK does not handle the event, because it was not sent by CIO. 

Are you suggesting renaming function names? Suggesting adding comments to explain something better?

Reply:
Yes, this is the behaviour I was expecting too, but naming suggested `expectEventHandled()` which makes me think why would CIO handle it.